### PR TITLE
Rules.mk Enhancements

### LINKED ISSUE #44: Support lower or mixed case source file and extensions
When I use lower case file extensions, BOB rules are not triggered.  When I use lowercase file names even with upper-case file extensions, the compile still doesn't happen.

### LINKED ISSUE #66: Remove the need to qualify source files with $(d) when specifying dependencies in Rules.mk
i.e instead of
COU300.MODULE: $(d)/COU300.RPGLE

just have

COU300.MODULE: COU300.RPGLE

### PR DESCRIPTION
**Pull request by TongkunZhang**
Fixes #66 
Absorbs #123 and Fixes #44

- Simplified Rules.mk Syntax
See example: https://github.com/edmundreinhardt/bob-recursive-example/pull/19

- Added indication for custom recipes
- Removed duplication definition of SQLs in header.mk

### CODE REVIEW

File: makei/init_project.py [hunk 1]
```diff
@@ -78,23 +78,6 @@ def _get_repository(self) -> str:
...
-            "repository": self.repository,
-            "includePath": self.include_path,
-            "objlib": self.objlib,
-            "curlib": self.curlib,
-            "preUsrLibl": self.pre_usr_libl,
-            "postUsrLibl": self.post_usr_libl,
-            "setIBMiEnvCmd": self.set_ibm_i_env_cmd
-        }
-        return json.dumps(iproj, indent=4)
-
```
Comment (TongkunZhang) [inline makei/init_project.py:97]: iproj_json.save()

**Review by TongkunZhang — COMMENTED** [review]
makei/crtfrmstmf.py file permission not correct

**Comment by edmundreinhardt** [comment]
Tongkun, have you made those 2 small changes from the review yesterday morning?

**Comment by TongkunZhang** [comment]
> Tongkun, have you made those 2 small changes from the review yesterday morning?

All the issues have been resolved. Some improvements have been added since the last review. The special keywords for the target variable are supported and tested.

File: mk/header.mk [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # Clear vars used by this make system
 define HEADER
 CLEAN :=
-TARGETS = $(TRGs) $(DTAs) $(SQLs) $(BNDDs) $(PFs) $(LFs) $(DSPFs) $(PRTFs) $(CMDs) $(SQLs) $(MODULEs) $(SRVPGMs) $(PGMs) $(MENUs) $(PNLGRPs) $(QMQRYs) $(WSCSTs) $(MSGs)
+TARGETS = $(TRGs) $(DTAs) $(SQLs) $(BNDDs) $(PFs) $(LFs) $(DSPFs) $(PRTFs) $(CMDs) $(MODULEs) $(SRVPGMs) $(PGMs) $(MENUs) $(PNLGRPs) $(QMQRYs) $(WSCSTs) $(MSGs)
```
Comment (edmundreinhardt) [inline mk/header.mk:4]: Just curious why you are dropping SQL from TARGETs?

**Review by edmundreinhardt — APPROVED** [review]

File: mk/header.mk (hunk 2 above)
Comment (TongkunZhang) [reply mk/header.mk:4]: `TARGETS = $(TRGs) $(DTAs) **$(SQLs)**`
Because there is a duplicate one at the begining

**Pull request merged**